python/pyarrow/includes/libarrow.pxd

```diff
@@ -366,6 +366,12 @@ cdef extern from "arrow/api.h" namespace "arrow" nogil:
         const uint8_t* GetValue(int i, int32_t* length)
 
     cdef cppclass CStringArray" arrow::StringArray"(CBinaryArray):
+        CStringArray(int64_t length, shared_ptr[CBuffer] value_offsets,
+                     shared_ptr[CBuffer] data,
```

Comment:
This is a stripped down-version without default arguments or const declarations as otherwise the compilation would fail with:

```
/home/uwe/Development/arrow-repos-3/arrow/python/build/temp.linux-x86_64-3.6/lib.cxx: In function ‘PyObject* __pyx_pf_7pyarrow_3lib_11StringArray_from_buffers(int, __pyx_obj_7pyarrow_3lib_Buffer*, __pyx_obj_7pyarrow_3lib_Buffer*, __pyx_obj_7pyarrow_3lib_Buffer*, int, int)’:
/home/uwe/Development/arrow-repos-3/arrow/python/build/temp.linux-x86_64-3.6/lib.cxx:42804:53: error: aggregate ‘__pyx_pf_7pyarrow_3lib_11StringArray_from_buffers(int, __pyx_obj_7pyarrow_3lib_Buffer*, __pyx_obj_7pyarrow_3lib_Buffer*, __pyx_obj_7pyarrow_3lib_Buffer*, int, int)::__pyx_opt_args_12CStringArray_CStringArray __pyx_t_5’ has incomplete type and cannot be defined
   struct __pyx_opt_args_12CStringArray_CStringArray __pyx_t_5;
```